src/libsyntax/parse/parser.rs

```diff
@@ -245,6 +245,7 @@ pub struct Parser<'a> {
     pub desugar_doc_comments: bool,
     /// Whether we should configure out of line modules as we parse.
     pub cfg_mods: bool,
+    allow_lifetimes: bool,
```

Comment:
yeah this doesn't feel very good

Reply:
it seems a bit confusing to me though -- we are specifically looking for lifetimes that come after a `&`, right? i.e., that's the only place a user might expect to have them. Can we just add a check to the parsing of `&` in an expression context for lifetimes? (And forego the rest?)